src/InsertionsClient/Api/Props/PropsVariableDeducer.cs

```diff
@@ -0,0 +1,175 @@
+            {
+                _credentials = new PackageSourceCredential(feed, accessToken, accessToken, true, null);
+            }
+        }
+
+        /// <summary>
+        /// Deduces the value of the variables in swr files
+        /// </summary>
+        /// <param name="defaultConfigUpdater"></param>
+        /// <param name="updatedPackages"></param>
+        /// <param name="swrFiles"></param>
+        /// <returns></returns>
+        public List<PropsFileVariableReference> DeduceVariableValues(DefaultConfigUpdater defaultConfigUpdater, IEnumerable<PackageUpdateResult> updatedPackages, SwrFile[] swrFiles)
+        {
+            ConcurrentBag<PropsFileVariableReference> deducedVariables = new ConcurrentBag<PropsFileVariableReference>();
+
+            TransformManyBlock<PackageUpdateResult, string> nugetDownloadBlock = new TransformManyBlock<PackageUpdateResult, string>(async (packageUpdate) =>
+            {
+                return await GetPackageFileList(defaultConfigUpdater, packageUpdate);
+            });
+
+            ActionBlock<string> filenameMatchBlock = new ActionBlock<string>((filename) =>
+            {
+                foreach (PropsFileVariableReference varReference in MatchFileNames(swrFiles, filename))
+                {
+                    deducedVariables.Add(varReference);
+                }
+            });
+
+            nugetDownloadBlock.LinkTo(filenameMatchBlock, new DataflowLinkOptions() { PropagateCompletion = true });
+
+            foreach (PackageUpdateResult package in updatedPackages)
+            {
+                nugetDownloadBlock.Post(package);
+            }
+
+            nugetDownloadBlock.Complete();
+            filenameMatchBlock.Completion.Wait();
+
+            return deducedVariables.ToList();
+        }
+
+        private async Task<IEnumerable<string>> GetPackageFileList(DefaultConfigUpdater defaultConfigUpdater, PackageUpdateResult packageUpdate)
+        {
+            string? link = defaultConfigUpdater.GetPackageLink(packageUpdate.PackageId);
+
+            if (link == null)
+            {
+                // We don't know where nuget package should be extracted to.
+                return Enumerable.Empty<string>();
+            }
+
+            SourceCacheContext cache = new SourceCacheContext();
+            CancellationTokenSource cancellationToken = new CancellationTokenSource();
+            NuGetVersion packageVersion = new NuGetVersion(packageUpdate.NewVersion);
+
+            // Stream that will receive the package bytes
+            using MemoryStream packageStream = new MemoryStream();
+
+            try
+            {
+                SourceRepository? rep = Repository.Factory.GetCoreV3(_feed, FeedType.HttpV3);
+                if (_credentials != null)
+                {
+                    rep.PackageSource.Credentials = _credentials;
+                }
+
+                FindPackageByIdResource? resource = await rep.GetResourceAsync<FindPackageByIdResource>();
```

Comment:
consider configuring the awaitable to false.

`await rep.GetResourceAsync<FindPackageByIdResource>().ConfigureAwait(false);`